docs/source/distribution_implementation.md

```diff
@@ -0,0 +1,452 @@
+from pymc3.distributions.dist_math import bound
+
+# Subclassing `PositiveContinuous` will dispatch a default `log` transformation
+class Blah(PositiveContinuous):
+
+    # This will be used by the metaclass `DistributionMeta` to dispatch the
+    # class `logp` and `logcdf` methods to the `blah` `op`
+    rv_op = blah
+
+    # dist() is responsible for returning an instance of the rv_op. We pass
+    # the standard parametrizations to super().dist
+    @classmethod
+    def dist(cls, param1, param2=None, alt_param2=None, **kwargs):
+        param1 = at.as_tensor_variable(intX(param1))
+        if param2 is not None and alt_param2 is not None:
+            raise ValueError('Only one of param2 and alt_param2 is allowed')
+        if alt_param2 is not None:
+            param2 = 1 / alt_param2
+        param2 = at.as_tensor_variable(floatX(param2))
+
+        # The first value-only argument should be a list of the parameters that
+        # the rv_op needs in order to be instantiated
+        return super().dist([param1, param2], **kwargs)
+
+    # Logp returns a symbolic expression for the logp evaluation of the variable
+    # given the `value` of the variable and the parameters `param1` ... `paramN`
+    def logp(value, param1, param2):
+        logp_expression = value * (param1 + at.log(param2))
+
+         # We use `bound` for parameter validation. After the default expression,
+         # multiple comma-separated symbolic conditions can be added. Whenever
+         # a bound is invalidated, the returned expression evaluates to `-np.inf`
+         return bound(
+            logp_expression,
+            value >= 0,
+            param2 >= 0,
+            # There is one sneaky optional keyowrd argument, that converts an
+            # otherwise elemwise `bound` to a reduced scalar bound. This is usually
+            # needed for multivariate distributions where the dimensionality
+            # of the bound conditions does not match that of the "value" / "logp"
+            # By default it is set to `True`.
+            broadcast_conditions=True,
+        )
+
+    # logcdf works the same way as logp. For bounded variables, it is expected
+    # to return `-inf` for values below the domain start and `0` for values
+    # above the domain end, but only when the parameters are valid.
+    def logcdf(value, param1, param2):
+        ...
+
+```
+
+Some notes:
+
+1. A distribution should at the very least inherit from `Discrete` or
+   `Continuous`. For the latter, more specific subclasses exist:
+   `PositiveContinuous`, `UnitContinuous`, `BoundedContinuous`,
+   `CircularContinuous`, which specify default transformations for the variables.
+   If you need to specify a one-time custom transform you can also override the
+   `__new__` method, as is done for the [`Dirichlet`](https://github.com/pymc-devs/pymc3/blob/6eaf9a0d7bc5b9168bedc9337808391ccb86a5ea/pymc3/distributions/multivariate.py#L399-L401).
```

Comment:
This could also be `` {class}`pymc3.Dirichlet` ``, but again, this goes to the api docs that will have the `[source]` link so it's two clicks, I'll try to find a better way. 

I don't like references to github lines because they either point to a specific commit and get outdated without anyone realizing so checking the link goes to the right code but that code is no longer in use within the actual library or they point to the main/dev branch and any change in line numbers makes the link point to the right file but wrong section.